app/controllers/banks_controller.rb

```diff
@@ -0,0 +1,18 @@
+class BanksController < ApplicationController
+
+  def receive_data
+    @basic_info = params[:basicInfo]
+    @contact_info = params[:contactInfo]
+
+    @bank = Bank.new
+    @contact_card = ContactCard.new
+
+    @bank.contact_card = @contact_card
+    @bank.from_json(@basic_info)
+    @contact_card.from_json(@contact_info)
+
+    @contact_card.save!
```

Comment:
You can put this code into a `Transaction` => http://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html

This way if anything fails inside a transaction, nothing will be saved